Statistics.cs

```diff
@@ -381,5 +415,73 @@ private void UpdateXMLFile(string cityName, ulong population, string fileName)
             cityPopulation.InnerText = population.ToString();
             document.Save("../../../Data/Canadacities-XML.xml");
         }
+        /// <summary>
+        /// Bool flag to tell whether the input from the user is correct.
+        /// </summary>
+        /// <param name="city">user input string.</param>
+        /// <returns>bool result</returns>
+        public bool CityIsValid(string city)
+        {
+            bool valid = false;
+            foreach (var item in CityCatalogue)
+            {
+                if (item.Value.GetCityName() == city) return true;
+            }
+            return valid;
+        }
+        /// <summary>
+        /// Bool flag to tell whether the input from the user is correct.
+        /// </summary>
+        /// <param name="city">user input string.</param>
+        /// <returns>bool result</returns>
+        public bool ProvinceIsValid(string city)
+        {
+            bool valid = false;
+            foreach (var item in CityCatalogue)
+            {
+                if (item.Value.GetProvince() == city) return true;
+            }
+            return valid;
+        }
+        /// <summary>
+        /// Displays the cities list.
+        /// </summary>
+        public void DisplayCitiesList()
+        {
+            string cities = "\n\n";
+            int counter = 0;
+            var items = from pair in CityCatalogue
+                        orderby pair.Key ascending
+                        select pair;
+
+
+            foreach (var item in items)
+            {
+                if (counter % 3 == 0)
+                {
+                    cities += "\n";
+                }
+                cities += "  " + item.Value.GetCityName().PadRight(30);
+                if (counter % 3 != 2)
+                    cities += ("|");
+                counter++;
+            }
+            Console.WriteLine(cities);
```

Comment:
Im not entirely sure if you have been having any performance issues, but considering we went over it in class, I dont think Hany would be too happy to see this. Try using a StringBuilder so it doesn't have a huge performance hit with reallocating strings.